### DISCUSSION TITLE
Accessibility review

### DISCUSSION THREAD

**Discussion by M-J-Robbins** [email-markup-consortium/email-markup-consortium#66] [General]
## Project title

Annual email accessibility review
## Project description

An annual accessibility review of emails sent, similar to the [WebAim million](https://webaim.org/projects/million/).

This idea came from an [accessibility review](https://twitter.com/M_J_Robbins/status/1518998121585553410) @avigoldman did for a webinar.

We need to collect a lot of sample emails from somewhere, hopefully we can do this in partnership with a company that already has a lot of data such as Really Good Emails or Mailcharts.  Also this project will overlap with a lot with [Email Platform Status](https://github.com/email-markup-consortium/email-markup-consortium/discussions/41) so a shared source of data would help both projects.

## Why is this needed

An accessibility review will highlight common issues, which we can then focus our efforts on. It will also help track progress across the industry.

## Scope

At first this project will only cover HTML email, but we may be open to other MIME types at a later date.

If historical data is easily available we could look at backdating a couple of years to help spot trends.

This will only cover automated testing and not manual testing.

## Contributing members

@M-J-Robbins @avigoldman 

## Where will the project be worked on, and where will the finished project be hosted?

Create a new repo under the EMC umbrella.
Results posted on EMC website blog

**Comment by Malvoz** [comment]
> An annual accessibility review of emails sent, similar to the [WebAim million](https://webaim.org/projects/million/).

The HTTP Archive's [Web Almanac](https://almanac.httparchive.org/en/2021/accessibility) is a great resource for inspiration (and info!) as well.

**Comment by husseinalhammad** [comment]
This project has been approved and will be led by @M-J-Robbins and @avigoldman 🎉

**Comment by M-J-Robbins** [comment]
So I thought I'd start making some notes on what we might want to include in the report.
I'm listing everything at first, we can cut this back to priority metrics if we don't have the time/resources to cover all of it.


* % of emails with critical issues
* % of emails with highest issues as serious
*  % of emails with highest issues as moderate
*  % of emails with highest issues as mild
* % of emails with no issues
* % if email without HTML 
* Top 10 most common issues
* Breakdown of the above by language (top 10 languages for which we have data)
* Breakdown the above by ESP used to send the email.
* Is it possible to detect frameworks by class names? If so add a breakdown by framework

This can be accompanied by some analysis and theories about why certain issues are more common. And from that we can look at some possible solutions to help fix these.

**Reply by husseinalhammad** [reply]
Does "sender" here mean ESP? Or the brand/entity that sends the email via an ESP? (Mailchimp vs company that uses Mailchimp)

**Reply by M-J-Robbins** [reply]
In this case, that would be the ESP that sent the email. I've edited it to make it more clear

**Reply by M-J-Robbins** [reply]
Additionally, we could pull out some of the email with 0 errors and run some manual test on them. Then add a section to the report that calls out brands that are doing an outstanding job.

**Comment by avigoldman** [comment]
Frameworks to look at:
* MJML
* Maizzle
* Cerberus Email
* Zurb

**Reply by husseinalhammad** [reply]
Given developers that use these frameworks are able to add their own code too, are we reviewing any emails built using these frameworks? Or just the HTML the frameworks include/generate by default?

**Reply by M-J-Robbins** [reply]
I think any email built with the framework.

The idea is we might find a conclusion like **_"People using [Framework] are likely to fail [Issue]"_**

We could then talk to the framework and see if that's something they can help their users with, maybe by adding a feature or adding a warning/recommendation.


Anything where we show breakdown by ESP or Framework or anything like that. We should make it very clear that this is not a reflection of the product but of how people use it.

**Comment by M-J-Robbins** [comment]
The results are in and the report is written (draft version). So next question is where do we post the results of the reviews/reports?

We're planning to do more of these reports so it would make sense to have a section on the site which holds them all.  This could live in the docs section so would be something like

emailmarkup.org > docs > reports > accessibility > July 2022

**Reply by husseinalhammad** [reply]
p.s. It doesn't need to be a subpage of `/en/docs` for it to appear in the docs section. e.g. it can be something like `/en/reports/accessibility` if needed.

Alternatively, we can set up projects pages which can have subpages e.g. `/en/docs/accessibility-reports` (background on the project, defines the general testing methodology, etc) and separate report pages like `/en/docs/accessibility-reports/2022-july`.
